x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/visualization_extractor.ts

```diff
@@ -93,14 +64,19 @@ export class VisualizationExtractor {
       );
     }
 
-    const fields = layer.accessors.map((a) => columns[a]);
+    const metricFields = layer.dimensions.filter((dimension) => dimension.role === 'metric');
 
-    const splitField = layer.splitAccessor ? columns[layer.splitAccessor] : null;
+    validateDimensions(metricFields);
+
+    const splitField = layer.dimensions.find(
```

Comment:
Could you please explain why it needed?

Reply:
It's not needed as such, it's just the pattern that's been used to indicate that `splitField` is explicitly not set.

Reply:
Undefined is good indicate too

Reply:
Ok, it's a pattern that we often use as `null` is a better indicator that a value is intentionally not set rather than being missing or unknown. 
But it's a matter of opinion and you can find lots of arguments for using either, so we can leave it as is.